engine/api/src/main/java/org/apache/cloudstack/engine/subsystem/api/storage/DataMotionStrategy.java

```diff
@@ -25,11 +25,28 @@
 import com.cloud.agent.api.to.VirtualMachineTO;
 import com.cloud.host.Host;
 
+/**
+ * interface to query how to move data around and to commision the moving
+ */
 public interface DataMotionStrategy {
+    /**
+     * reports whether this instance can do a move from source to destination
+     * @param srcData object to move
+     * @param destData location to move it to
+     * @return the expertise level with which this instance knows how to handle the move
+     */
     StrategyPriority canHandle(DataObject srcData, DataObject destData);
 
     StrategyPriority canHandle(Map<VolumeInfo, DataStore> volumeMap, Host srcHost, Host destHost);
 
+    /**
+     * copy the source volume to its destination (on a host if not null)
```

Comment:
what happens if the host is null?

Is it possible to be null when we get to this part of the code?

Reply:
maybe not in this PR but it is part of the bigger framework for data motion.

Reply:
@rafaelweingartner StorageMotionStrategy that supports host to be null is selected and used for migrations, e.g. VmwareStorageStrategyMotion used by this PR/feature.

A non-blank host is only used for copying objects between primary storage pools.